InfrastructureManager/agents/vsphere_agent.py

```diff
@@ -0,0 +1,104 @@
+"""
+[TESTING]
+Agent class for vSphere.
+The agent need pysphere 0.1.8 you can install from pip or from esayinstall tools
+
+You need setup your vCenter:
+	you need create a AppScale template with VMWare tools installed and to be powered on.
+  	you need create a head node in vSphere
+"""
+
+from agents.base_agent import BaseAgent, AgentConfigurationException, AgentRuntimeException
+import datetime
+from pysphere import VIServer
+import time
+from utils import utils
+
+__author__ = 'Marco Maria Di Nicola'
+__email__ = 'marticus@hotmail.it'
+
+PARAM_CREDENTIALS = 'credentials'           #HOST(vCenter ip), USERNAME (vCenter), PASSWORD (vCenter)
+PARAM_IMAGE_ID = 'image_id'                 #name of AppScale Template create in vSphere some like appscale-template
+PARAM_RESOURCE_POOL = 'resource_pool'       #name of Resouce Pool in vSphere (reservate for AppScale) some like appscale-pool
+PARAM_INSTANCE_IDS = 'instance_ids'         #name of vm (ex: appscale1)
+
+class vSphereAgent(BaseAgent):
+	"""
+	vSphere infrastructure agent class which can be used to spawn and terminate
+  	VMs in an vSphere based environment.
+	"""
+	def configure_instance_security(self, parameters):
+		"""
+		Prepare AppScale head node in vCenter and AppScale template for cloning
+		a new virtual machines.
+		"""
+		return True
+
+	def open_connection(self, parameters):
+		"""
+		Initialize a connection to the back-end vSphere.
+
+	    Args:
+	      parameters  A dictionary containing the 'credentials' parameter.
+
+	    Returns:
+	      An instance of pysphere connection.
+		"""
+		credentials = parameters[PARAM_CREDENTIALS]
+		try:
+			server = VIServer()
+			server.connect(credentials['HOST'], credentials['USERNAME'], credentials['PASSWORD'])
+		except StandardError, e:
+			print ('Connection problem with the host {0}'.format(credentials['HOST']))
```

Comment:
Please use the logger for printing messages throughout this file, like so: https://github.com/AppScale/appscale/blob/master/InfrastructureManager/infrastructure_manager.py#L182